src/mappers/bittrex.ts

```diff
@@ -0,0 +1,150 @@
+import { inflateRawSync } from 'zlib'
+import { BookChange, Exchange, Trade } from '../types'
+import { Mapper } from './mapper'
+
+export class BittrexOrderChangeMapper implements Mapper<'bittrex', BookChange> {
+  constructor(protected readonly exchange: Exchange) {}
+
+  canHandle(message: BittrexMessageType) {
+    return isValid(message, 'orderBook') || message.stream === 'depthSnapshot'
+  }
+
+  getFilters(symbols?: string[]) {
+    return [
+      {
+        channel: 'depthSnapshot',
+        symbols
+      } as const,
+      {
+        channel: 'orderBook',
+        symbols
+      } as const
+    ]
+  }
+
+  *map(message: BittrexMessageType, localTimestamp: Date): IterableIterator<BookChange> {
+    // the response between message from snapshot service and message from signalr are different
+    let bookChange: BookChange
+    if (message.data) {
+      const data = message.data
+      bookChange = {
+        type: 'book_change',
+        symbol: message.symbol!,
+        exchange: this.exchange,
+        isSnapshot: true,
+        bids: data.bid.map(this.mapBookLevel),
+        asks: data.ask.map(this.mapBookLevel),
+        timestamp: message.timestamp!,
+        localTimestamp
+      }
+    } else {
+      const data: BittrexOrderChange = fetchAndParseMessage(message)
+      bookChange = {
+        type: 'book_change',
+        symbol: data.marketSymbol,
+        exchange: this.exchange,
+        isSnapshot: false,
+        bids: data.bidDeltas.map(this.mapBookLevel),
+        asks: data.askDeltas.map(this.mapBookLevel),
+        timestamp: localTimestamp,
+        localTimestamp
+      }
+    }
+    yield bookChange
+  }
+
+  private mapBookLevel(level: BittrexBookLevel) {
+    const amount = Number(level.quantity)
+    const price = Number(level.rate)
+    return { price, amount }
+  }
+}
+
+export class BittrexTradesMapper implements Mapper<'bittrex', Trade> {
+  constructor(private readonly exchange: Exchange) {}
+
+  canHandle(message: BittrexMessageType) {
+    return isValid(message, 'trade')
+  }
+
+  getFilters(symbols?: string[]) {
+    return [
+      {
+        channel: 'trade',
+        symbols
+      } as const
+    ]
+  }
+
+  *map(message: any, localTimestamp: Date): IterableIterator<Trade> {
```

Comment:
should it be `message: BittrexMessageType` ?

Reply:
Yep, you are right